src/LiquidationBot.ts

```diff
@@ -254,24 +257,32 @@ export default class LiquidationBot {
 
   async run() {
     const users = await this.computeLiquidableUsers();
-    const liquidationsParams = await Promise.all(
-      users.map((u) => this.getUserLiquidationParams(u.address))
-    );
-    const toLiquidate = (
-      await Promise.all(
-        liquidationsParams.map(async (user) => {
-          if (
-            await this.isProfitable(
-              user.debtMarket.market,
-              user.toLiquidate,
-              user.debtMarket.price
+    this.logger.log(`Found ${users.length} users liquidatable`);
+    // use the batch size to limit the number of users to liquidate
+    const toLiquidate: UserLiquidationParams[] = [];
+    for (let i = 0; i < users.length; i += this.settings.batchSize) {
+      const liquidationsParams = await Promise.all(
+        users
+          .slice(i, Math.min(i + this.settings.batchSize, users.length))
+          .map((u) => this.getUserLiquidationParams(u.address))
+      );
+      const batchToLiquidate = (
+        await Promise.all(
+          liquidationsParams.map(async (user) => {
+            if (
+              await this.isProfitable(
+                user.debtMarket.market,
+                user.toLiquidate,
+                user.debtMarket.price
+              )
             )
-          )
-            return user;
-          return null;
-        })
-      )
-    ).filter(Boolean);
+              return user;
+            return null;
+          })
+        )
+      ).filter(Boolean) as UserLiquidationParams[];
+      toLiquidate.push(...batchToLiquidate);
+    }
```

Comment:
IMO, Would be better with a `Promise.all` and a `flat` at the end, to highly parallelise the Promise. Or is it done voluntarily ? 🙂

Reply:
volontary to prevent RPC throughput errors